src/App.tsx

```diff
@@ -1,26 +1,313 @@
+  const [allDone, setAllDone] = useState(false);
+  const [IsSubmitting, setIsSubmitting] = useState(false);
+  const [tempTodo, setTempTodo] = useState<TempTodo | null>(null);
+  const [deletingIds, setDeletingIds] = useState<number[]>([]);
+  const [isUpdating, setIsUpdating] = useState<number[] | []>([]);
+  const [canEdit, setCanEdit] = useState(false);
+  const completedTodos = tasks?.filter(todo => todo.completed);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const errorType: ErrorType = {
+    empty: 'Title should not be empty',
+    load: 'Unable to load todos',
+    add: 'Unable to add a todo',
+    found: 'Todo not found',
+    deleteTask: 'Unable to delete a todo',
+    updateTodo: 'Unable to update a todo',
+  };
+  const taskLeft = tasks?.filter(
+    task => !task.completed && task.id != 0,
+  ).length;
+
+  const focusInput = () => {
+    setTimeout(() => {
+      inputRef.current?.focus();
+    }, 0);
+  };
+
+  const handleStatus = (value: Status) => {
+    setStatus(value);
+  };
+
+  const handleNewTaskTitle = (eventTarget: string) => {
+    setTaskTitle(eventTarget);
+  };
+
+  const handleSetTasks = (newTasks: Todo[]) => {
+    setTasks(newTasks);
+  };
+
+  const handleIsSubmitting = (isSubmitting: boolean) => {
+    setIsSubmitting(isSubmitting);
+  };
+
+  const handleEdit = (edit: boolean) => {
+    setCanEdit(edit);
+  };
+
+  const filterTodo: (todos: Todo[], mode: Status) => Todo[] = (todos, mode) => {
+    switch (mode) {
+      case Status.completed:
+        return todos.filter(task => task.completed);
+      case Status.active:
+        return todos.filter(task => !task.completed);
+      case Status.all:
+      default:
+        return tasks;
+    }
+  };
+
+  const filteredTodos = filterTodo(tasks, status);
+
+  const handleError: (errorMsg: string) => void = errorMsg => {
+    setErrorMessage(errorMsg);
+  };
+
+  const handleErrorClear = () => {
+    setErrorMessage('');
+  };
+
+  const addNewTodo = async (creatNewTodo: Todo) => {
+    try {
+      setTasks(currentTodos => [...currentTodos, creatNewTodo]);
+      setIsUpdating([0]);
+
+      const newTodo: Todo = await addTodo(creatNewTodo);
+
+      setTasks(currentTodos => {
+        currentTodos.pop();
+
+        return [...currentTodos, newTodo];
+      });
+      setTaskTitle('');
+    } catch {
+      setTasks(currentTodos => {
+        currentTodos.pop();
+
+        return [...currentTodos];
+      });
+      handleError(errorType.add);
+    } finally {
+      setTempTodo(null);
+      setIsSubmitting(false);
+      focusInput();
+    }
+  };
+
+  const handleSubmit = (event: { preventDefault: () => void }) => {
+    event.preventDefault();
+    const title = taskTitle.trim();
+
+    if (!title) {
+      handleError(errorType.empty);
+
+      return;
+    }
+
+    const newTodo = {
+      title,
+      userId: USER_ID,
+      completed: false,
+      id: 0,
+    };
+
+    setIsSubmitting(true);
+    addNewTodo(newTodo);
+  };
```

Comment:
Move this function to the `Header.tsx` component and manage the `taskTitle` state there. Pass `handleError` and `addNewTodo` as props to the Header component. Additionally, remove `setIsSubmitting` and set `isSubmitting` to true directly inside the `addNewTodo` function.